compiler-utils/src/main/java/com/squareup/anvil/compiler/internal/reference/TypeReference.kt

```diff
@@ -128,23 +128,23 @@ public sealed class TypeReference {
     override val unwrappedTypes: List<Psi> by lazy(NONE) {
       type.typeElement!!.children
         .filterIsInstance<KtTypeArgumentList>()
-        .single()
-        .children
-        .filterIsInstance<KtTypeProjection>()
-        .map { typeProjection ->
+        .singleOrNull()
+        ?.children
+        ?.filterIsInstance<KtTypeProjection>()
+        ?.map { typeProjection ->
           typeProjection.children
             .filterIsInstance<KtTypeReference>()
             .single()
             .toTypeReference(declaringClass)
-        }
+        } ?: emptyList()
     }
 
-    override fun resolveGenericTypeOrNull(implementingClass: ClassReference): TypeReference? {
+    override fun resolveGenericTypeOrNull(implementingClass: ClassReference): TypeReference {
```

Comment:
The new return type with this method name is kinda weird. I think I'd keep a nullable return type and just suppress the warning in the IDE.